docs/design/INTEGRATE-ADMINAPI.V1.md

```diff
@@ -0,0 +1,488 @@
+
+# Integration Design: EdFi.Ods.AdminApi.V1 with EdFi.Ods.AdminApi (V2)
+
+## Overview
+
+This document outlines the design for integrating EdFi.Ods.AdminApi.V1 endpoints
+into the existing EdFi.Ods.AdminApi (V2) solution. The integration will maintain
+backward compatibility while leveraging the enhanced architecture of V2.
+
+### Goals
+
+* Maintain backward compatibility for existing V1 API clients
+* Leverage V2's enhanced architecture and infrastructure
+* Minimize code duplication between versions
+* Provide clear migration path for V1 to V2
+* Centralize common functionality
```

Comment:
These are great goals. Reflection on each.

* _Maintain backward compatibility for existing V1 API clients_ --> HIGHEST priority. I'm trusting that the existing Postman tests give good coverage on this.
* _Leverage V2's enhanced architecture and infrastructure_ --> MEDIUM priority. It is not necessary to do much refactoring.
* _Minimize code duplication between versions_ --> LOW priority at first - yes it is good to reduce code duplication, but it will not be necessary to do that all at one time. We can gradually reduce the tech debt from duplicated code.
* _Provide clear migration path for V1 to V2_ --> yes
* _Centralize common functionality_ --> again low priority _at first_, maybe a higher priority as time goes by.